src/core/ext/client_channel/client_channel.c

```diff
@@ -236,12 +239,11 @@ static void on_resolver_result_changed(grpc_exec_ctx *exec_ctx, void *arg,
     lb_policy_args.client_channel_factory = chand->client_channel_factory;
 
     // Find LB policy name.
-    const char *lb_policy_name = NULL;
     const grpc_arg *channel_arg =
         grpc_channel_args_find(lb_policy_args.args, GRPC_ARG_LB_POLICY_NAME);
     if (channel_arg != NULL) {
       GPR_ASSERT(channel_arg->type == GRPC_ARG_STRING);
-      lb_policy_name = channel_arg->value.string;
+      lb_policy_name = gpr_strdup(channel_arg->value.string);
```

Comment:
Good idea.  Unfortunately, I can't do the `gpr_strdup()` on line 307, because the value on this line is being assigned from an argument contained in `chand->resolver_result`, which we're destroying before we leave this block (which ends on line 297).  However, I've moved it down to just before we destroy `chand->resolver_result`, with an appropriate comment.